### PR TITLE
cmake/modules/BuildRocksDB.cmake: enable compressions for rocksdb

### PR DESCRIPTION
**Pull request by tchaikov**
we should enable them if they are found.

currently, we don't have bzip2 compressor plugin, so it's not detected
in the cmake script. we can always enable it for rocksdb in future.

Fixes: http://tracker.ceph.com/issues/24025
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by socketpair** [comment]
@tchaikov thanks for the invitation, but unfortunatelly, I am not familiar with build system of Ceph.

File: cmake/modules/BuildRocksDB.cmake [hunk 1]
```diff
@@ -13,6 +13,9 @@ function(do_build_rocksdb)
     list(APPEND ROCKSDB_CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER})
   endif()
 
+  list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_SNAPPY=${SNAPPY_FOUND})
+  list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_LZ4=${LZ4_FOUND})
+  list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_ZLIB=${ZLIB_FOUND})
   list(APPEND ROCKSDB_CMAKE_ARGS -DPORTABLE=ON)
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply cmake/modules/BuildRocksDB.cmake]: as the commit message explains. we don't detect bzip2 and bzip2 is not linked by default. so i'm inclined to not do so in this change. Zstd is compiled as a compressor plugin, but rocksdb detects libzstd as an existing library. we need to 1) make zstd a dependency of rocksdb 2) skip the `find_package(zstd REQUIRED)` step in rocksdb's cmake script 3) pass `ZSTD_INCLUDE_DIR` and `ZSTD_LIBRARIES` down to rocksdb's cmake script. this requires the change in rocksdb. so i'd like to hold this change to another PR if necessary.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2018-05-24-1437/

**Pull request merged**